### ISSUE TITLE
Setting RabbitMQ headers and properties

### ISSUE THREAD

**Issue by karthik20522** [vanthoainguyen/Burrow.NET#20]
Is there any example on how to set headers and properties while publishing a message. I would like to custom add headers and properties. For example in official rabbitmq library, we can do the following:

'''
Dictionary<string, object> headers = new Dictionary<string, object>();
headers.Add("category", "helloworld");
IBasicProperties basicProperties = model.CreateBasicProperties();
basicProperties.Headers = headers;
byte[] messageBytes = Encoding.UTF8.GetBytes(message);
 model.BasicPublish(_headersExchange, "", basicProperties, messageBytes);
'''

Basically I would like custom headers similar to above example. Any guidance would be much appreciated.

**Comment by vanthoainguyen** [comment]
Hi,

The Tunnel object has a method to publish message with a customer header:

```csharp
/// <summary>
/// Use this method to publish a message with custome headers to a "headers" exchange defined by route finder. 
/// <para>The message will be published to that exchange with a routing key resolved by the route finder</para>
/// <para>So if the target exchange is Direct or Fanout, the customHeaders have no value here and should be ignored by RabbitMQ. </para>
/// <para>This method should be used when you're sure that you're publishing to a "headers" exchange</para>
/// </summary>
/// <typeparam name="T"></typeparam>
/// <param name="rabbit">A message to publish</param>
/// <param name="customHeaders">If you're publishing to a Header exchange, this method allows you to put your header parameters</param>
void Publish<T>(T rabbit, IDictionary<string, object> customHeaders);
```

So you will simply need to prepare the dictionary and use it.
